platform/ios/MGLMapView.mm

```diff
@@ -3324,6 +3328,40 @@ void afterRender() override
         const float scaleFactor;
 };
 
+
+class MGLLogObserver : public mbgl::Log::Observer
+{
+public:
+    MGLLogObserver(MGLMapView *nativeView_)
+        : nativeView(nativeView_) {}
+    virtual ~MGLLogObserver() {}
+    
+    bool onRecord(mbgl::EventSeverity severity, mbgl::Event event, int64_t code, const std::string &msg)
+    {
+        __block bool didNotifyDelegate = false;
+        if (severity == mbgl::EventSeverity::Error && event == mbgl::Event::Setup)
```

Comment:
Currently, it’s only used inside `MapContext::setStyleURL()`, when the style itself fails to load due to a failed HTTP request. Is that consistent with what you outlined in https://github.com/mapbox/mapbox-gl-native/issues/2775#issuecomment-150829090?